src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DefaultBindingMetadataProvider.cs

```diff
@@ -53,6 +54,21 @@ public void GetBindingMetadata([NotNull] BindingMetadataProviderContext context)
                 context.BindingMetadata.PropertyBindingPredicateProvider = new CompositePredicateProvider(
                     predicateProviders);
             }
+
+            if (context.Key.MetadataKind == ModelMetadataKind.Property)
+            {
+                // BindingBehavior can fall back to attributes on the Container Type, but we should ignore
+                // attributes on the Property Type.
+                var bindingBehavior =
+                    context.PropertyAttributes.OfType<BindingBehaviorAttribute>().FirstOrDefault() ??
+                    context.Key.ContainerType.GetTypeInfo().GetCustomAttribute<BindingBehaviorAttribute>(inherit: true);
```

Comment:
@harshgMSFT this is the current behavior of this attribute, nothing has changed. I added functional tests for this in a previous PR.